sys/suit/handlers_command_seq.c

```diff
@@ -424,7 +424,10 @@ static int _dtv_fetch(suit_manifest_t *manifest, int key,
         return err;
     }
 
-    assert(manifest->urlbuf && url_len < manifest->urlbuf_len);
+    if (!manifest->urlbuf || url_len >= manifest->urlbuf_len) {
+        assert(0);
+        return -ENOBUFS;
```

Comment:
```
/**
 * @brief SUIT error codes
 */
typedef enum {
    SUIT_OK                   =  0, /**< Manifest parsed and validated */
    SUIT_ERR_INVALID_MANIFEST = -1, /**< Unexpected CBOR structure detected */
    SUIT_ERR_UNSUPPORTED      = -2, /**< Unsupported SUIT feature detected */
    SUIT_ERR_NOT_SUPPORTED    = -3, /**< Unsupported features detected */
    SUIT_ERR_COND             = -4, /**< Conditionals evaluate to false */
    SUIT_ERR_SEQUENCE_NUMBER  = -5, /**< Sequence number less or equal to
                                         current sequence number */
    SUIT_ERR_SIGNATURE        = -6, /**< Unable to verify signature */
    SUIT_ERR_DIGEST_MISMATCH  = -7, /**< Digest mismatch with COSE and SUIT */
    SUIT_ERR_POLICY_FORBIDDEN = -8, /**< Denied because of policy mismatch */
    SUIT_ERR_NO_MEM           = -9, /**< Out of memory condition */
```

```suggestion
        return SUIT_ERR_NO_MEM;
```
    SUIT_ERR_STORAGE          = -50, /**< Backend returned an error */
    SUIT_ERR_STORAGE_EXCEEDED = -51, /**< Backend out of space */
    SUIT_ERR_STORAGE_UNAVAILABLE = -52, /**< Backend location not available */
} suit_error_t;